client/gutenberg/extensions/simple-payments/edit.js

```diff
@@ -90,7 +128,7 @@ class SimplePaymentsEdit extends Component {
 				spay_multiple: multiple,
 				spay_price: price,
 			},
-			status: 'publish',
+			status: paymentId ? 'publish' : 'draft',
```

Comment:
Mostly this works well, although there is one important consideration.

We serialize this as a shortcode 😕 

That would be fine if the shortcode respected the product post status as I expected, but it does not appear to 😞 

I'm going to look at Jetpack to verify this, but the best approach may be to file a PR with Jetpack to ensure we only render products from shortcodes if _the product is published_.

---

A more subtle note: By looking at this line, it's not clear what's happening or why. Here's what I understand to be happening. It would be great if this were more explicit.

When we add a brand new button, we call this method. In that case, we don't have a `paymentId`, so we create a draft. This is necessary so that as we work on the post and change the simple payment, we can save the post and the simple payment concurrently.

As the payment is modified, values are stored as block attributes. When the containing post is saved, we attempt to save the product if it is valid. In that case, we would update the product's status to `publish`.

---

Reply:
https://github.com/Automattic/jetpack/blob/6ee14cffce0947b7ddfdeb0b58bdda5b7e7edb35/modules/simple-payments/simple-payments.php#L119-L121

```php
		if ( $product->post_type !== self::$post_type_product || 'trash' === $product->post_status ) {
			return;
		}
```

If it's not in the trash, show it 🤷‍♂️